gridstatus/ieso.py

```diff
@@ -0,0 +1,473 @@
+
+        url = LOAD_TEMPLATE_URL.replace(
+            "YYYYMMDDHH",
+            f"{(date).strftime('%Y%m%d')}{hour:02d}",
+        )
+
+        r = self._request(url, verbose)
+
+        root = ET.fromstring(r.text)
+
+        # Extracting all triples of Interval, Market Total Load, and Ontario Load
+        interval_loads_and_demands = self._find_loads_at_each_interval_from_xml(root)
+
+        df = pd.DataFrame(
+            interval_loads_and_demands,
+            columns=["Interval", "Market Total Load", "Ontario Load"],
+        )
+
+        delivery_date = root.find("DocBody/DeliveryDate", NAMESPACES_FOR_XML).text
+        delivery_hour = int(root.find("DocBody/DeliveryHour", NAMESPACES_FOR_XML).text)
+        created_at = root.find("DocHeader/CreatedAt", NAMESPACES_FOR_XML).text
+
+        df["Delivery Date"] = pd.Timestamp(delivery_date, tz=self.default_timezone)
+
+        # The starting hour is 1, so we subtract 1 to get the hour in the range 0-23
+        df["Delivery Hour Start"] = delivery_hour - 1
+        # Multiply the interval minus 1 by 5 to get the minutes in the range 0-55
+        df["Interval Minute Start"] = MINUTES_INTERVAL * (df["Interval"] - 1)
+        df["Published Time"] = pd.Timestamp(created_at, tz=self.default_timezone)
+
+        df["Interval Start"] = (
+            df["Delivery Date"]
+            + pd.to_timedelta(df["Delivery Hour Start"], unit="h")
+            + pd.to_timedelta(df["Interval Minute Start"], unit="m")
+        )
+
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(
+            minutes=MINUTES_INTERVAL,
+        )
+
+        cols_to_keep_in_order = [
+            "Interval Start",
+            "Interval End",
+            "Published Time",
+            "Market Total Load",
+            "Ontario Load",
+        ]
+
+        df = utils.move_cols_to_front(df, cols_to_keep_in_order)[cols_to_keep_in_order]
+
+        if end:
+            return df[df["Interval End"] <= pd.Timestamp(end)]
+
+        return df
+
+    def get_load(self, date, end=None, verbose=False):
+        """
+        Get hourly load for the Market and Ontario for a given date or from
+        date to end date.
+
+        Args:
+            date (datetime.date | datetime.datetime | str): The date to get the load for
+                Can be a `datetime.date` or `datetime.datetime` object, or a string
+                with the values "today" or "latest". If `end` is None, returns
+                only data for this date.
+            end (datetime.date | datetime.datetime, optional): End date. Defaults None
+                If provided, returns data from `date` to `end` date. The `end` can be a
+                `datetime.date` or `datetime.datetime` object.
+            verbose (bool, optional): Print verbose output. Defaults to False.
+
+
+        Returns:
+            pd.DataFrame: zonal load as a wide table with columns for each zone
+        """
+        if isinstance(date, tuple):
```

Comment:
you should be able to avoid having to handle tuples this by passing `date` and `end` through to a function that is decorated with `@support_date_range`. like eventually these params will hit `_retrieve_5_minute_load` which should handle this for you

Reply:
This method doesn't support a date range though since there's only one load forecast file available.

Reply:
got it. in that case, i think the only allowed values for `date` should be `today` and `latest`.  and then i would remove `end` from the function signature

Reply:
Updated.